### PR TITLE
Update container-jfr image version

### PR DESCRIPTION
**Pull request by andrewazores**
Also update controllers for updated connectionless API, currently
emulating a connection-oriented interface by holding internal state.
Finally, also replaces "ALL" event string with "template=ALL".

### CODE REVIEW

**Comment by andrewazores** [comment]
The (dis)connect methods on the `JfrClient` aren't really necessary anymore and therefore the `ClientLock` mutex shouldn't be, either, but I've left things as similar as possible here intentionally to minimize the risk of any regressions.

File: deploy/olm-catalog/container-jfr-operator-bundle/0.4.0/container-jfr-operator-bundle.v0.4.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ metadata:
             "archive": true,
             "duration": "30s",
             "eventOptions": [
-              "ALL"
+              "template=ALL"
```
Comment (ebaron) [inline deploy/olm-catalog/container-jfr-operator-bundle/0.4.0/container-jfr-operator-bundle.v0.4.0.clusterserviceversion.yaml:41]: Perhaps we should leave the 0.4.0 CSV as-is, and do a new "release" with version bump and updated CSV?

Comment (andrewazores) [reply deploy/olm-catalog/container-jfr-operator-bundle/0.4.0/container-jfr-operator-bundle.v0.4.0.clusterserviceversion.yaml:41]: Sure, I'll back out that part of the change. I think you did the last CSV update - do you want to do this next one too?

Comment (ebaron) [reply deploy/olm-catalog/container-jfr-operator-bundle/0.4.0/container-jfr-operator-bundle.v0.4.0.clusterserviceversion.yaml]: Sure, I can do that.

**Pull request merged**